### ISSUE TITLE
Kubernetes API server unreachable from some containers

### ISSUE THREAD

**Issue by localghost** [rancher/rancher#8543]
**Rancher Versions:**
Server: 1.5.3
healthcheck: 0.2.3
ipsec: 0.9.4
network-services: 0.9.1
scheduler: k8s:v1.5.4-rancher1-3
kubernetes (if applicable): 1.5.4-rancher1

**Docker Version:**
1.12.6

**OS and where are the hosts located? (cloud, bare metal, etc):**
CentOS 7.2 on KVM

**Setup Details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
single node rancher

**Environment Type: (Cattle/Kubernetes/Swarm/Mesos)**
Kubernetes

**Steps to Reproduce:**
1. Run Rancher server
2. Create Kubernetes environment
3. Add host (run Rancher agent on the same node as Rancher server)

All the steps are done via Ansible scripts using Rancher's V2 API.

**Results:** 
Kubernetes Dashboard does not launch _for some time_, container repeatedly starts and fails with error:
```
4/19/2017 9:08:04 AMUsing HTTP port: 9090
4/19/2017 9:08:04 AMCreating API server client for https://10.43.0.1:443
4/19/2017 9:08:34 AMError while initializing connection to Kubernetes apiserver. This most likely means that the cluster is misconfigured (e.g., it has invalid apiserver certificates or service accounts configuration) or the --apiserver-host param points to a server that does not exist. Reason: Get https://10.43.0.1:443/version: dial tcp 10.43.0.1:443: i/o timeout
4/19/2017 9:08:34 AMRefer to the troubleshooting guide for more information: https://github.com/kubernetes/dashboard/blob/master/docs/user-guide/troubleshooting.md
```
I have checked all the things suggested on [Kubernetes debug service](https://kubernetes.io/docs/tasks/debug-application-cluster/debug-service/) and K8s API server seems to be setup properly, i.e. service is deployed, endpoint for the service exists, iptables rules seem to be correct. API server responds when I query it from host using either its cluster/service IP or endpoint IP. However, from within containers (which don't have network mode set to _host_) API server is reachable only via endpoint IP, querying it using its cluster IP results in timeout (like in case of Kubernetes Dashboard above).

What's even more puzzling is that after some time, like 10-15 minutes, API server starts responding on the cluster IP and Dashboard kicks in.

I've checked logs of lb-controller, kubernetes agent, kube proxy, etc but I haven't found anything suspicious, but since I started using Rancher quite recently I might be missing something.

**Comment by jralmaraz** [comment]
Same happened to me a couple of times, but I could not precise what happened. Do you have any update on this? I have created new environments from scratch and in some of them I did not observer this behavior and some yes.

**Comment by localghost** [comment]
No, no update from my side. The only thing that comes to my mind is that since API server starts responding on cluster IP after some time maybe there is some cache in one of the Rancher's component involved.

**Comment by wasi** [comment]
I'm having the same issue with 1.5.4-rancher-1-2

**Comment by localghost** [comment]
The issue seems to be fixed in rancher 1.5.4. Though, I am not sure which bugfix/change resolved the issue.
